### DISCUSSION TITLE
How to remove all styles / api.delete()

### DISCUSSION THREAD

**Discussion by mutheusalmeida** [pmndrs/react-spring#1993] [Support] [answered]
Hey, guys. I have an animation that works in a way on mobile devices and in another on desktop. How can I remove all previous styles conditionally and start again with new ones when the viewport change? Is it with the `api.delete()` method? I cound'nt figure out how to use the delete method, I would love to see an example of what it receives

Thanks in advance

**Comment by joshuaellis** [comment, marked as answer]
If you want to "reset" all styles, I would use the `set` method to set the current values back to defaults and possible set up your new values ready for the `start` method to animate.

**Reply by mutheusalmeida** [reply]
How can I start the animation again after calling set?

**Reply by joshuaellis** [reply]
> start method to animate.

https://beta.react-spring.dev/docs/advanced/spring-ref#start
